### PR TITLE
[CIS-648] Audit underlying channel list views

### PR DESCRIPTION
**Pull request by mantergo**
I was a bit slowed down because of the #824 so here is the 1st part of the CIS-648 to see if I'm doing this audit correct.

**In this PR:**
- Audit `ChatChannelReadStatusCheckmarkView`
- Audit `ChatChannelUnreadCountView`
- Audit `ChatChannelSwipeableListItemView`

### CODE REVIEW

File: Sources/StreamChatUI/ChatChannelList/ChatChannelUnreadCountView.swift [hunk 1]
```diff
@@ -5,63 +5,34 @@
...
 import UIKit
 
+/// A view that shows a number of unread messages in channel.
 public typealias ChatChannelUnreadCountView = _ChatChannelUnreadCountView<NoExtraData>
-
+/// A view that shows a number of unread messages in channel.
 open class _ChatChannelUnreadCountView<ExtraData: ExtraDataTypes>: View, UIConfigProvider {
-    // MARK: - Properties
+    /// The `UILabel` instance that holds number of unread messages.
+    open private(set) lazy var unreadCountLabel = UILabel().withoutAutoresizingMaskConstraints
```
Comment (VojtaStavik) [inline Sources/StreamChatUI/ChatChannelList/ChatChannelUnreadCountView.swift:13]: this should probably also have `adjustingFontForContentSizeCategory` -> https://github.com/GetStream/stream-chat-swift/pull/819

File: Sources/StreamChatUI/ChatChannelList/ChatChannelSwipeableListItemView.swift [hunk 2]
```diff
@@ -73,14 +74,10 @@ open class _ChatChannelSwipeableListItemView<ExtraData: ExtraDataTypes>: View, U
         deleteButton.tintColor = uiConfig.colorPalette.alert
     }
 
-    // MARK: - Button actions
-
-    @objc func didTapDelete() {
+    @objc private func didTapDelete() {
         deleteButtonAction?()
     }
```
Comment (VojtaStavik) [inline Sources/StreamChatUI/ChatChannelList/ChatChannelSwipeableListItemView.swift:79]: why `private` when what you were looking for was `open`? :D

File: Tests/StreamChatUITests/Helpers.swift [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+// Copyright © 2021 Stream.io Inc. All rights reserved.
+//
+
+import UIKit
+
+extension UIView {
+    /// Simulate embedding in superview to run view lifecycle methods.
+    func embedInSuperview() {
+        UIView().addSubview(self)
+    }
```
Comment (VojtaStavik) [inline Tests/StreamChatUITests/Helpers.swift:11]: Maybe we should call this `simulateAddingToSuperview()` or even `executeLifecycleMethods()`?

File: Sources/StreamChatUI/ChatChannelList/ChatChannelSwipeableListItemView_Tests.swift [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+        let view = ChatChannelSwipeableListItemView().withoutAutoresizingMaskConstraints
+        
+        view.uiConfig = config
+        view.addSizeConstraints()
+
+        // Simulate view moving to superview to have view initialized.
+        view.embedInSuperview()
+        
+        // Buttons are revealed after swipe gesture so we need to simulate it to check the buttons.
+        view.trailingConstraint?.constant = 50
```
Comment (VojtaStavik) [inline Sources/StreamChatUI/ChatChannelList/ChatChannelSwipeableListItemView_Tests.swift:37]: Can we create some test-only extension for this like `simulateOpen()` or `swipeOpen()`?

**Review by VojtaStavik — APPROVED** [review]
Cool, looks great 👍 Just some minor comments regarding naming and API.

File: Sources/StreamChatUI/ChatChannelList/ChatChannelUnreadCountView.swift (hunk 1 above)
Comment (mantergo) [reply Sources/StreamChatUI/ChatChannelList/ChatChannelUnreadCountView.swift:13]: It was before audit already (line 43) 🙂

File: Sources/StreamChatUI/ChatChannelList/ChatChannelUnreadCountView.swift [hunk 5]
```diff
@@ -5,63 +5,34 @@
...
     
-    public var inset: CGFloat = 3
+    /// A `CGFloat`value that defines insets for embedding `unreadCountLabel`.
+    open private(set) var inset: CGFloat = 3
     
     override open var intrinsicContentSize: CGSize {
         let height: CGFloat = max(unreadCountLabel.font.pointSize + inset * 2, frame.height)
         let width = max(unreadCountLabel.intrinsicContentSize.width + inset * 2, height)
         return .init(width: width, height: height)
     }
```
[the remark this answers is not in the document]
Comment (mantergo) [reply Sources/StreamChatUI/ChatChannelList/ChatChannelUnreadCountView.swift:22]: I've tried that and this thing needs to be really dynamic to work with adjustable fonts and inside reusable cells so autolayout approach was having issues and made things harder to understand so I think it's the best option for this view.

**Pull request merged**